router/broker.go

```diff
@@ -505,56 +595,24 @@ func (b *broker) syncPubEvent(pub *wamp.Session, msg *wamp.Publish, pubID wamp.I
 			}
 		}
 
-		// TODO: Handle publication trust levels
-
-		details := wamp.Dict{}
-		if eventDetails != nil {
-			details = eventDetails
-		}
+		event := prepareEvent(pub, msg, pubID, sub, sendTopic, disclose, eventDetails, subscriber)
+		b.trySend(subscriber, event)
+	}
 
-		event := &wamp.Event{
-			Publication:  pubID,
-			Subscription: sub.id,
-			Arguments:    msg.Arguments,
-			ArgumentsKw:  msg.ArgumentsKw,
-			Details:      details,
-		}
-		// If a subscription was established with a pattern-based matching
-		// policy, a Broker MUST supply the original PUBLISH.Topic as provided
-		// by the Publisher in EVENT.Details.topic|uri.
-		if sendTopic {
-			event.Details[detailTopic] = msg.Topic
-		}
-		if disclose && subscriber.HasFeature(wamp.RoleSubscriber, wamp.FeaturePubIdent) {
-			disclosePublisher(pub, event.Details)
+	// If event history store is enabled for subscription let's save event
+	if eventStore, ok := b.eventHistoryStore[sub]; ok {
+		// We should ignore publications that has exclude or eligible lists of session IDs
+		// As session ID is runtime thing, it is impossible to check later: is history asker allowed to get
```

Comment:
```suggestion
		// As session ID is a runtime thing, it is impossible to check later: is history asker allowed to get
```

In addition `asker` reads bad, perhaps re-phrase this?

Reply:
👍🏻